setupmeta/__init__.py

```diff
@@ -36,7 +36,7 @@ def abort(message):
 
 def warn(message):
     """Issue a warning (coming from setupmeta itself)"""
-    warnings.warn(message)
+    warnings.warn(message, stacklevel=2)
```

Comment:
Today I learned: the existence of `stacklevel` :)
Thank you, much better this way!